x/mongo/driver/session/client_session.go

```diff
@@ -215,16 +215,22 @@ func NewClientSession(pool *Pool, clientID uuid.UUID, sessionType Type, opts ...
 		return nil, errors.New("causal consistency and snapshot cannot both be set for a session")
 	}
 
-	servSess, err := pool.GetSession()
-	if err != nil {
-		return nil, err
+	if sessionType == Explicit {
```

Comment:
Can we leave a comment above this explaining why we only `SetServer` here for `Explicit` sessions and `Implicit` sessions call `SetServer` after connection checkout in `Execute`?

Reply:
Absolutely.